src/snowflake/snowpark/mock/plan.py

```diff
@@ -1406,14 +1442,42 @@ def calculate_expression(
                     else:
                         res_cols.append(None)
             res_col = ColumnEmulator(
-                data=res_cols, dtype=object
+                data=res_cols,
+                dtype=object,
+                sf_type=calculate_expression(
+                    window_function.expr,
+                    windows[0],
+                    analyzer,
+                    expr_to_alias,
+                    keep_literal=True,
+                ).sf_type,
             )  # dtype=object prevents implicit converting None to Nan
             res_col.index = res_index
             return res_col.sort_index()
         else:
             raise NotImplementedError(
                 f"[Local Testing] Window Function {window_function} is not implemented."
             )
+    elif isinstance(exp, SubfieldString):
+        col = calculate_expression(exp.child, input_data, analyzer, expr_to_alias)
+        field = str(exp.field)
+        # in snowflake, two consecutive single quotes means escaping single quote
+        field = field.replace("''", "'")
```

Comment:
Question: Where does the quote get escaped?

Reply:
singe quote is escaped if it's a string constant: https://docs.snowflake.com/en/sql-reference/data-types-text#single-quoted-string-constants

in this PR, it's the test
```python
    def variant2(cls, session: "Session") -> DataFrame:
        return session.sql(
            """
            select parse_json(column1) as src
            from values
            ('{
                "date with '' and ." : "2017-04-28",
                "salesperson" : {
                  "id": "55",
                  "name": "Frank Beasley"
                },
                "customer" : [
                  {"name": "Joyce Ridgely", "phone": "16504378889", "address": "San Francisco, CA"}
                ],
                "vehicle" : [
                  {"make": "Honda", "extras":["ext warranty", "paint protection"]}
                ]
            }')
            """
```

Reply:
probably we need to apply this escape logic when parsing the string literal